agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/configuration/ConfigurationBuilder.java

```diff
@@ -201,6 +202,18 @@ private static void logConfigurationWarnings(Configuration config) {
                 + " \"spanKind\" to \"telemetryKind\".");
       }
     }
+    if (!config.preview.instrumentationKeyOverrides.isEmpty()) {
+      configurationLogger.warn(
+          "Instrumentation key overrides have been deprecated,"
+              + " and support for it will be removed in a future release, please transition from"
+              + " \"instrumentationKeyOverrides\" to \"connectionStringOverrides\".");
+      for (InstrumentationKeyOverride override : config.preview.instrumentationKeyOverrides) {
+        ConnectionStringOverride newOverride = new ConnectionStringOverride();
```

Comment:
i see, it's applied to connectionstring override. just keeping config backward compatible, but underneath, it's all connectionstring override now.